resources/channel.sub.js

```diff
@@ -0,0 +1,1003 @@
+        let objectMap = new Map();
+
+        while (stack.length > 0) {
+            const {item, target, targetKey} = stack.shift();
+            const {type, value, objectId} = item;
+            let result;
+            let newTarget;
+            if (objectId !== undefined && value === undefined) {
+                result = objectMap.get(objectId);
+            } else {
+                switch(type) {
+                case "undefined":
+                    result = undefined;
+                    break;
+                case "null":
+                    result = null;
+                    break;
+                case "string":
+                case "boolean":
+                    result = value;
+                    break;
+                case "number":
+                    if (typeof value === "string") {
+                        switch(value) {
+                        case "NaN":
+                            result = NaN;
+                            break;
+                        case "-0":
+                            result = -0;
+                            break;
+                        case "+Infinity":
+                            result = Number.POSITIVE_INFINITY;
+                            break;
+                        case "-Infinity":
+                            result = Number.NEGATIVE_INFINITY;
+                            break;
+                        default:
+                            throw new Error(`Unexpected number value "${value}"`);
+                        }
+                    } else {
+                        result = value;
+                    }
+                    break;
+                case "bigint":
+                    result = BigInt(value);
+                    break;
+                case "function":
+                    result = new Function("...args", `return (${value}).apply(null, args)`);
+                    break;
+                case "remoteobject":
+                    let remote = new RemoteObject(value.type, value.objectId);
+                    let local = remote.toLocal();
+                    if (local !== null) {
+                        result = local;
+                    } else {
+                        result = remote;
+                    }
+                    break;
+                case "sendchannel":
+                    result = new SendChannel(value);
+                    break;
+                case "regexp":
+                    result = new RegExp(value.pattern, value.flags);
+                    break;
+                case "date":
+                    result = new Date(value);
+                    break;
+                case "error":
+                    if (item.value.type in self &&
+                        typeof self[item.value.type] === "function") {
+                        result = new self[item.value.type](item.value.message);
+                    } else {
+                        result = new Error(item.value.message);
```

Comment:
If `item.value.type` is "Error", the above case should be taken. Can you add a comment that this isn't about deserializing `Error` objects, but non-`Error` values?